R/diagnosis_helper_functions.R

```diff
@@ -208,3 +208,83 @@ clean_bootstrap_df <- function(diagnosis, digits, diagnosand_columns,
 
   return(return_df)
 }
+
+
+
+#' Title
+#'
+#' @param diagnosis 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+tidy.diagnosis <- function(diagnosis) {
+  
+  diagnosand_columns <- diagnosis$diagnosand_names
+  
+  diagnosands_df <- get_diagnosands(diagnosis)
+  
+  parameter_names <- names(diagnosis$parameters_df)[-1]
+  
+  group_by_set <- diagnosis$group_by_set
+  
+  if (is.data.frame(diagnosis$bootstrap_replicates)) {
+    diagnosand_se_columns <- paste0("se(", diagnosis$diagnosand_names, ")")
+    group_columns <- setdiff(names(diagnosands_df), c(diagnosand_columns, diagnosand_se_columns))
+    
+    diagnosand_replicates <- diagnosis$bootstrap_replicates
+    
+    group_by_list <- diagnosand_replicates[, group_by_set, drop = FALSE]
+    
+    labels_df <- split(group_by_list, lapply(group_by_list, addNA), drop = TRUE)
+    labels_df <- lapply(labels_df, head, n = 1)
+    
+    use_vars <- setdiff(names(diagnosand_replicates), c(group_by_set, "bootstrap_id"))
+    diagnosands_summary_df <- split(diagnosand_replicates[use_vars], lapply(group_by_list, addNA), drop = TRUE)
+    
+    diagnosands_mean_list <- lapply(diagnosands_summary_df, function(data) lapply(data[diagnosand_columns], mean))
```

Comment:
I think either a helper function or for-loop would be appropriate here, easier to follow / modify later compared to 4x nested lapplys + unlists.

Reply:
That is fair. :) Will fix.